frontend/src/Frontend/UI/DeploymentSettings.hs

```diff
@@ -251,14 +252,19 @@ buildDeploymentSettingsResult m mSender signers cChainId capabilities ttl gasLim
   code' <- lift code
   keys <- lift $ m ^. wallet_keys
   allAccounts <- MaybeT $ Map.lookup networkName . unAccountStorage <$> m ^. wallet_accounts
-  -- Make an effort to ensure the sender account has enough balance to actually
+
+  -- Make an effort to ensure the gas payer (if there is one) account has enough balance to actually
   -- pay the gas. This won't work if the user selects an account on a different
   -- chain, but that's another issue.
-  for_ (lookupAccountBalance sender allAccounts) $ \case
-    Nothing -> empty
-    Just b -> let GasLimit lim = _pmGasLimit publicMeta
-                  GasPrice (ParsedDecimal price) = _pmGasPrice publicMeta
-               in guard $ unAccountBalance b > fromIntegral lim * price
+  gasPayer <- lift mGasPayer
+  case gasPayer of
+    Nothing -> pure () -- No gas payer selected, move along
+    Just gp ->  for_ (lookupAccountBalance gp allAccounts) $ \case
+      Nothing -> empty -- Gas Payer selected but they're not an account?!
```

Comment:
Should we show some UI feedback in this case?

Reply:
Added some feedback and refactored this function to provide verbose error information if more is required in future.